### DISCUSSION TITLE
Can I improve this pattern on the compiler.

### DISCUSSION THREAD

**Discussion by hrasoa** [webpack/webpack#15078] [Q&A]
Hello, I'd like some advice about this pattern. I would like to execute something once the ```serverCompiler``` has done its first compile. I tried many hooks but I could find the one that suites my need. As you can see I ended up putting a flag.

```js
const clientServer = new webpackDevServer(clientDevServerOptions, clientCompiler);  

let init;

clientServer.start().then(() => {
    const serverCompiler = webpack(serverConfig);
    serverCompiler.hooks.done.tap('ondone', (stats) => {
        if (!init) {
            init = true;
            // do something once
        }
    })
    serverCompiler.watch({} , (err, stats) => {});
});
```

**Comment by alexander-akait** [comment]
Please move `const serverCompiler = webpack(serverConfig);` before `clientServer.start()`
